### PR TITLE
Add appendRawHistoryNodes api

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
1. Add new persistence API to append raw history batches
2. GetWorkflowExecutionRawHistoryV2 return node IDs 

**Why?**
Allow us to apply raw history events.

**How did you test it?**
Integration test

**Potential risks**

**Is hotfix candidate?**
No

### CODE REVIEW

File: common/persistence/tests/history_store.go [hunk 1]
```diff
@@ -81,7 +84,8 @@ func NewHistoryEventsSuite(
 			logger,
 			dynamicconfig.GetIntPropertyFn(4*1024*1024),
 		),
-		logger: logger,
+		serializer: serialization.NewSerializer(),
```
Comment (yycptt) [inline common/persistence/tests/history_store.go:87]: nit: use the same serializer as the one passed to `NewExecutionManager`?

File: common/persistence/dataInterfaces.go [hunk 2]
```diff
@@ -719,6 +719,26 @@ type (
...
+		// the info for clean up data in background
+		Info string
+		// The branch to be appended
+		BranchToken []byte
+		// The batch of events to be appended. The first eventID will become the nodeID of this batch
+		History *commonpb.DataBlob
+		// TransactionID for events before these events. For events chaining
+		PrevTransactionID int64
+		// requested TransactionID for this write operation. For the same eventID, the node with larger TransactionID always wins
+		TransactionID int64
```
Comment (yycptt) [inline common/persistence/dataInterfaces.go:737]: nit: not sure if it worth creating a common struct for Append and AppendRaw and embed it in those two requests.

**Review by yycptt — APPROVED** [review]

**Review by yiminc — APPROVED** [review]

File: common/persistence/dataInterfaces.go (hunk 2 above)
Comment (yux0) [reply common/persistence/dataInterfaces.go:737]: Good point

Comment (yux0) [reply common/persistence/dataInterfaces.go:737]: Cannot think of a good name and structure. May do it in another PR.

**Pull request merged**